contracts/contracts/harvest/OETHHarvesterSimple.sol

```diff
@@ -0,0 +1,107 @@
+    /// --- STORAGE
+    ////////////////////////////////////////////////////
+    address public operator;
+    address public strategist;
+    mapping(address => bool) public isAuthorized;
+
+    ////////////////////////////////////////////////////
+    /// --- EVENTS
+    ////////////////////////////////////////////////////
+    event Harvested(address token, uint256 amount);
+    event OperatorSet(address operator);
+    event StrategistSet(address strategist);
+    event StrategyStatusSet(address strategy, bool status);
+
+    ////////////////////////////////////////////////////
+    /// --- MODIFIERS
+    ////////////////////////////////////////////////////
+    modifier onlyOperator() {
+        require(
+            msg.sender == operator || isGovernor(),
+            "Only Operator or Governor"
+        );
+        _;
+    }
+
+    ////////////////////////////////////////////////////
+    /// --- CONSTRUCTOR
+    ////////////////////////////////////////////////////
+    constructor(
+        address _governor,
+        address _operator,
+        address _strategist
+    ) {
+        require(_strategist != address(0), "Invalid strategist");
+        operator = _operator;
+        strategist = _strategist;
+        _setGovernor(_governor);
+    }
+
+    ////////////////////////////////////////////////////
+    /// --- MUTATIVE FUNCTIONS
+    ////////////////////////////////////////////////////
+    function harvestAndTransfer(address _strategy) external onlyOperator {
+        _harvestAndTransfer(_strategy);
+    }
+
+    function harvestAndTransfer(address[] calldata _strategies)
+        external
+        onlyOperator
+    {
+        for (uint256 i = 0; i < _strategies.length; i++) {
+            _harvestAndTransfer(_strategies[i]);
+        }
+    }
+
+    function _harvestAndTransfer(address _strategy) internal {
+        // Ensure strategy is authorized
+        require(isAuthorized[_strategy], "Strategy not authorized");
+
+        // Harvest rewards
+        IStrategy(_strategy).collectRewardTokens();
+
+        // Cache reward tokens
+        address[] memory rewardTokens = IStrategy(_strategy)
+            .getRewardTokenAddresses();
+        for (uint256 i = 0; i < rewardTokens.length; i++) {
+            // Cache balance
+            uint256 balance = IERC20(rewardTokens[i]).balanceOf(address(this));
+            if (balance > 0) {
+                // Transfer to strategist
+                IERC20(rewardTokens[i]).transfer(strategist, balance);
+                emit Harvested(rewardTokens[i], balance);
+            }
+        }
+    }
+
+    ////////////////////////////////////////////////////
+    /// --- GOVERNANCE
+    ////////////////////////////////////////////////////
+    function setOperator(address _operator) external onlyGovernor {
+        operator = _operator;
+        emit OperatorSet(_operator);
+    }
+
+    function setStrategyStatus(address _strategy, bool _status)
+        external
+        onlyGovernor
```

Comment:
Maybe this could be strategist only?

This does mean that the strategist could make this contract make arbitrary calls to `IERC20(address).transfer(strategist, balance);`, but that shouldn't do any damage, given that this contract is basicly just an extension of the strategist powers in the first place, all funds go to the strategist anyway, and this should not be holding money or user funds?

Reply:
Agree! Changed in this commit: [23a885f](https://github.com/OriginProtocol/origin-dollar/pull/2333/commits/23a885f542f3dcc22d9a661d269ff382aa5cf19a)